tools/backup_analysis.py

```diff
@@ -0,0 +1,174 @@
+def check_backup(mydir):
+    if not os.path.isdir(mydir):
+        return False
+    backup_file=mydir + "/WiredTiger.backup"
+    if not os.path.exists(backup_file):
+        return False
+    return True
+
+def compare_file(dir1, dir2, filename, file_size, cmp_size, granularity):
+    # Initialize all of our counters per file.
+    bytes_gran = 0
+    gran_blocks = 0
+    num_blocks = file_size // cmp_size
+    partial = file_size % cmp_size
+    pct20 = granularity // 5
+    pct80 = pct20 * 4
+    pct20_count = 0
+    pct80_count = 0
+    start_off = 0
+    total_bytes_diff = 0
+    fp1 = open(os.path.join(dir1, filename), "rb")
+    fp2 = open(os.path.join(dir2, filename), "rb")
+    # Time how long it takes to compare each file.
+    start = time.asctime()
+    # Compare the bytes in cmp_size blocks between both files.
+    for b in range(0, num_blocks + 1):
+        offset = cmp_size * b
+        # Gather and report information when we cross a granularity boundary.
+        if offset % granularity == 0:
+            if offset != 0 and bytes_gran != 0:
+                print("{}: Offset {}: {} bytes differ in {} bytes".format(filename, start_off, bytes_gran, granularity))
+            # Account for small or large block changes.
+            if bytes_gran != 0:
+                if bytes_gran <= pct20:
+                    pct20_count += 1
+                elif bytes_gran >= pct80:
+                    pct80_count += 1
+            # Reset for the next granularity block.
+            start_off = offset
+            bytes_gran = 0
+
+        # Compare the two blocks. We know both files are at least file_size so all reads should work.
+        buf1 = fp1.read(cmp_size)
+        buf2 = fp2.read(cmp_size)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += cmp_size
+            # Count how many granularity level blocks changed.
+            if bytes_gran == 0:
+                gran_blocks += 1
+            bytes_gran += cmp_size
+    # Account for anything from the final iteration of the loop and partial blocks.
+    if partial != 0:
+        buf1 = fp1.read(partial)
+        buf2 = fp2.read(partial)
+        # If they're different, gather information.
+        if buf1 != buf2:
+            total_bytes_diff += partial
+            bytes_gran += partial
+    fp1.close()
+    fp2.close()
+    end = time.asctime()
+    if bytes_gran != 0:
+        if offset != 0:
+            print("{}: Offset {}: {} bytes differ in {} bytes".format(filename, start_off, bytes_gran, granularity))
+        if bytes_gran <= pct20:
+            pct20_count += 1
+        elif bytes_gran >= pct80:
+            pct80_count += 1
+
+    # Report for each file.
+    print("{}: started {} completed {}".format(filename, start, end))
+    print("{} ({}) differs by {} bytes in {} granularity blocks".format(filename, file_size, total_bytes_diff, gran_blocks ))
```

Comment:
The output could be more easily parsed with a simple change. Make all the lines start: file: word ... where work tells what type of line it is.  Now an example of the various lines is:

new.wt: Offset 0: 4096 bytes differ in 16777216 bytes
new.wt: started Fri Feb  2 00:15:45 2024 completed Fri Feb  2 00:15:45 2024
new.wt (4096) differs by 4096 bytes in 1 granularity blocks
new.wt: 1 of 1 blocks (100%) differ by 3355443 bytes or less of 16777216 (20%)
new.wt: 0 of 1 blocks (0%) differ by 13421772 bytes or more of 16777216 (80%)

To make it parse better change to:
new.wt: Offset 0: 4096 bytes differ in 16777216 bytes
new.wt: started Fri Feb  2 00:15:45 2024 completed Fri Feb  2 00:15:45 2024
new.wt: size 4096, differs by 4096 bytes in 1 granularity blocks
new.wt: Below 20%: 1 of 1 blocks (100%) differ by 3355443 bytes or less of 16777216
new.wt: Above 80% 0 of 1 blocks (0%) differ by 13421772 bytes or more of 16777216

The second word identifies the line type:
Offset - Range of different bytes
started - Time
size - Total difference summary
Below - Bottom 20%
Above - Top 20%

Reply:
Done.